recipes/professor/0001-Fix-LC_ID_DYLIB-on-macOS.patch

```diff
@@ -0,0 +1,25 @@
+From 18480a41cfb23cdabf1241cb20ea9e793b4349b2 Mon Sep 17 00:00:00 2001
+From: Chris Burr <christopher.burr@cern.ch>
+Date: Fri, 4 Nov 2022 16:49:35 +0100
+Subject: [PATCH] Fix LC_ID_DYLIB on macOS
+
+---
+ Makefile | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/Makefile b/Makefile
+index baf73e2..e8043f7 100644
+--- a/Makefile
++++ b/Makefile
+@@ -105,7 +105,7 @@ lib: lib/libProfessor2.so
+ 
+ lib/libProfessor2.so: $(LIBOBJECTS)
+ 	mkdir -p lib
+-	$(CXX) -shared -o $@ $(LIBOBJECTS)
++	cd lib && $(CXX) -shared -o $(notdir $@) $(addprefix ../,$(LIBOBJECTS))
```

Comment:
Is this macOS specific or should it go into upstream?

Reply:
This can go upstream. There are better ways to do it but this is the least invasive change which will make it work on most unix-like systems.

Reply:
ok. I will include it in my MR targeting 2.3.5